Core/include/Acts/Geometry/ConeVolumeBounds.hpp

```diff
@@ -0,0 +1,228 @@
+  ~ConeVolumeBounds() override = default;
+
+  ConeVolumeBounds& operator=(const ConeVolumeBounds& cobo) = default;
+
+  VolumeBounds::BoundsType type() const final { return VolumeBounds::eCone; }
+
+  /// Return the bound values as dynamically sized vector
+  ///
+  /// @return this returns a copy of the internal values
+  std::vector<double> values() const final;
+
+  /// This method checks if position in the 3D volume
+  /// frame is inside the cylinder
+  ///
+  /// @param pos is the position in volume frame to be checked
+  /// @param tol is the absolute tolerance to be applied
+  bool inside(const Vector3D& pos, double tol = 0.) const final;
+
+  /// Oriented surfaces, i.e. the decomposed boundary surfaces and the
+  /// according navigation direction into the volume given the normal
+  /// vector on the surface
+  ///
+  /// @param transform is the 3D transform to be applied to the boundary
+  /// surfaces to position them in 3D space
+  ///
+  /// It will throw an exception if the orientation prescription is not adequate
+  ///
+  /// @return a vector of surfaces bounding this volume
+  OrientedSurfaces orientedSurfaces(
+      const Transform3D* transform = nullptr) const override;
+
+  /// Construct bounding box for this shape
+  /// @param trf Optional transform
+  /// @param envelope Optional envelope to add / subtract from min/max
+  /// @param entity Entity to associate this bounding box with
+  /// @return Constructed bounding box
+  Volume::BoundingBox boundingBox(const Transform3D* trf = nullptr,
+                                  const Vector3D& envelope = {0, 0, 0},
+                                  const Volume* entity = nullptr) const final;
+
+  /// Access to the bound values
+  /// @param bValue the class nested enum for the array access
+  double get(BoundValues bValue) const { return m_values[bValue]; }
+
+  // Return the derived innerRmin
+  double innerRmin() const;
+
+  // Return the derived innerRmin
+  double innerRmax() const;
+
+  // Return the derived inner tan(alpha)
+  double innerTanAlpha() const;
+
+  // Return the derived outerRmin
+  double outerRmin() const;
+
+  // Return the derived outerRmax
+  double outerRmax() const;
+
+  // Return the derived outer tan(alpha)
+  double outerTanAlpha() const;
+
+  /// Output Method for std::ostream
+  ///
+  /// @param sl is ostream operator to be dumped into
+  std::ostream& toStream(std::ostream& sl) const override;
+
+ private:
+  /// Check the input values for consistency,
+  /// will throw a logic_exception if consistency is not given
+  void checkConsistency() noexcept(false);
+
+  /// Create the surface bounds
+  void buildSurfaceBounds();
+
+  /// Templated dump methos
+  /// @tparam stream_t The type of the stream for dumping
+  /// @param dt The stream object
+  template <class stream_t>
+  stream_t& dumpT(stream_t& dt) const;
+
+  /// The bound values
+  std::array<double, eSize> m_values;
+  std::shared_ptr<CylinderBounds> m_innerCylinderBounds{nullptr};
```

Comment:
Could these be stored as values instead of pointers?

Reply:
Not in the current model, which relies on the fact that bounds are/can be shared between volumes - I am planning a local test where I change the voll geometry model to a pure private ownership (not as values as we do have some polymorphism).

In the current implementation, however, some of those are optional and it would steer how the `orientedSurfaces(...)` method behaves.